features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/APIDefinition/APIDefinition.jsx

```diff
@@ -0,0 +1,276 @@
+        let parsedContent = {};
+        if (this.hasJsonStructure()) {
+            parsedContent = JSON.parse(this.state.swagger);
+        } else {
+            try {
+                parsedContent = yaml.load(this.state.swagger);
+            } catch (err) {
+                Alert.error('Error while updating the API Definition');
+                return;
+            }
+        }
+
+        const promise = this.api.updateSwagger(parsedContent);
+        promise.then(
+            (response) => {
+                Alert.success('API Definition Updated Successfully');
+                this.closeEditor();
+            }
+        ).catch((err) => {
+            console.debug(err);
+            Alert.error('Error while updating the API Definition');
+        });
+    }
+
+    /**
+     * Checks whether the swagger content is json type.
+     * */
+    hasJsonStructure() {
+        const swagger = this.state.swagger;
+        if (typeof swagger !== 'string') return false;
+        try {
+            const result = JSON.parse(swagger);
+            return Object.prototype.toString.call(result) === '[object Object]'
+                || Array.isArray(result);
+        } catch (err) {
+            return false;
+        }
+    }
+
+    /**
+     * Handles the file upload.
+     * */
+    onDrop(file) {
+        if (file[0]) {
+            let reader = new FileReader();
+            reader.onload = (e) => {
+                let content = e.target.result;
+                this.setState({swagger: content}, () => {
+                    this.updateSwaggerDefinition()
+                });
+            }
+            reader.readAsText(file[0]);
+        } else {
+            Alert.error('Unsupported file type.')
+        }
+    }
+
+    render() {
+        const {swagger} = this.state;
+        const {classes} = this.props;
+
+        const editorOptions = {
+            selectOnLineNumbers: true,
+            readOnly: true,
+            smoothScrolling: true,
+            wordWrap: 'on',
+            handleMouseWheel: true,
+            scrollbar: {
+                handleMouseWheel: true
+            }
+        };
+        if (this.state.notFound) {
+            return <ResourceNotFound message={this.props.resourceNotFountMessage}/>;
+        }
+        if (!swagger) {
+            <Progress />
+        }
+
+        return (
+            <div className={classes.root}>
+                <div className={classes.titleWrapper}>
+                    <Typography variant='h4' align='left' className={classes.mainTitle}>
+                        API Definition
+                    </Typography>
+                    <Button size='small' className={classes.button} onClick={this.openEditor}>
+                        <EditRounded className={classes.buttonIcon}/>
+                        Edit
+                    </Button>
+                    <Dropzone
+                        multiple={false}
+                        className={classes.dropzone}
```

Comment:
Shall we destructure all the class name variables coming from MUI in the top of the render method?

i:e

```javascript
const { button, dropzone, buttonIcon } = classes
```

so we don't need to repeat classes.* in all the places